edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Diagnostics/Metric.cs

```diff
@@ -6,15 +6,16 @@ namespace Microsoft.Azure.Devices.Edge.Agent.Diagnostics
     using System.Collections.Generic;
     using System.Text;
     using Microsoft.Azure.Devices.Edge.Util;
+    using Newtonsoft.Json;
 
     public sealed class Metric : IEquatable<Metric>
     {
         public DateTime TimeGeneratedUtc { get; }
         public string Name { get; }
         public double Value { get; }
-        public string Tags { get; }
+        public Dictionary<string, string> Tags { get; }
```

Comment:
```suggestion
        public IReadonlyDictionary<string, string> Tags { get; }
```

Reply:
To be more safe, you can copy items from given dictionary in constructor.  To ensure, these values cannot be changed.